app/src/main/java/ru/woodymsk/socialapp/di/NavigationModule.kt

```diff
@@ -0,0 +1,28 @@
+package ru.woodymsk.socialapp.di
+
+import com.github.terrakok.cicerone.Cicerone
+import com.github.terrakok.cicerone.NavigatorHolder
+import com.github.terrakok.cicerone.Router
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+
+@InstallIn(SingletonComponent::class)
+@Module
+class NavigationModule {
+    private val cicerone: Cicerone<Router> = Cicerone.create()
+
+    @Provides
+    @Singleton
+    fun provideRouter(): Router {
+        return cicerone.router
+    }
```

Comment:
Так красивее будет. И ниже то же самое.

```suggestion
    fun provideRouter(): Router = cicerone.router
```

Reply:
исправил, внизу тоже